### DISCUSSION TITLE
How can I change the format of class names generated by CSS Modules in Next.js?

### DISCUSSION THREAD

**Discussion by lumenwrites** [vercel/next.js#37196] [Help]
Is there any way to customize the format of class names created for me by CSS modules?

Right now it looks like `[component name]_[class name]_[hash]`, and I want it to be just `[class name]_[shorter hash]`.

**Comment by GabenGar** [comment]
Classnames are generated by `css-loader` from webpack.
So you have to read how to extend [webpack config in nextjs](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config).
Then what changes to apply to [`css-loader` ](https://webpack.js.org/loaders/css-loader/#modules).

**Comment by MEddarhri** [comment]
@lumenwrites  you could do that by adding this webpack configuration to your next.config.js

```JavaScript
const nextConfig = {
 // your configuration...
  webpack: (
    config,
    { buildId, dev, isServer, defaultLoaders, nextRuntime, webpack }
  ) => {
    if (isServer === false) {
      config.module.rules.push({
        test: /\.css$/i,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              modules: {
                localIdentName: "[path][name]__[local]",
              },
            },
          },
          "sass-loader",
        ],
      });
    }
    return config;
  },
// your configuration...
};
```
And then you could set the classname by entering the format in  **localIdentName** like you want

**Reply by GabenGar** [reply]
Pretty sure `module.rules.push()` won't cut it since there is already a nextjs config for `.css` files, so it will double-compile the same files.

**Reply by Mifaresss** [reply]
Not working: nextjs overrides classNames
